execution-engine/engine-grpc-server/tests/test_stored_contract_support.rs

```diff
@@ -0,0 +1,922 @@
+        let mut protocol_version = ProtocolVersion::new();
+        protocol_version.set_value(version);
+        self.exec_request.set_protocol_version(protocol_version);
+        self
+    }
+
+    pub fn build(mut self) -> ExecuteRequest {
+        let mut deploys: protobuf::RepeatedField<DeployItem> =
+            <protobuf::RepeatedField<DeployItem>>::new();
+        for deploy in self.deploys {
+            deploys.push(deploy);
+        }
+        self.exec_request.set_deploys(deploys);
+        self.exec_request
+    }
+}
+
+impl Default for ExecRequestBuilder {
+    fn default() -> Self {
+        let deploys = vec![];
+        let mut exec_request = ExecuteRequest::new();
+        exec_request.set_block_time(DEFAULT_BLOCK_TIME);
+        let mut protocol_version = ProtocolVersion::new();
+        protocol_version.set_value(1);
+        exec_request.set_protocol_version(protocol_version);
+        ExecRequestBuilder {
+            deploys,
+            exec_request,
+        }
+    }
+}
+
+pub fn get_protocol_version() -> ProtocolVersion {
+    let mut protocol_version: ProtocolVersion = ProtocolVersion::new();
+    protocol_version.set_value(1);
+    protocol_version
+}
+
+pub fn get_mock_deploy() -> DeployItem {
+    let mut deploy = DeployItem::new();
+    deploy.set_address(MOCKED_ACCOUNT_ADDRESS.to_vec());
+    deploy.set_motes_transferred_in_payment(1000);
+    deploy.set_gas_price(1);
+    deploy.set_nonce(1);
+    let mut deploy_code = DeployCode::new();
+    deploy_code.set_code(test_utils::create_empty_wasm_module_bytes());
+    let mut deploy_payload = DeployPayload::new();
+    deploy_payload.set_deploy_code(deploy_code);
+    deploy.set_session(deploy_payload);
+    deploy
+}
+
+fn get_compiled_wasm_path(contract_file: PathBuf) -> PathBuf {
+    let mut path = std::env::current_dir().expect("should get working directory");
+    path.push(PathBuf::from(COMPILED_WASM_PATH));
+    path.push(contract_file);
+    path
+}
+
+/// Reads a given compiled contract file from [`COMPILED_WASM_PATH`].
+pub fn read_wasm_file_bytes(contract_file: &str) -> Vec<u8> {
+    let contract_file = PathBuf::from(contract_file);
+    let path = get_compiled_wasm_path(contract_file);
+    std::fs::read(path.clone())
+        .unwrap_or_else(|_| panic!("should read bytes from disk: {:?}", path))
+}
+
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum SystemContractType {
+    Mint,
+    ProofOfStake,
+}
+
+#[allow(clippy::implicit_hasher)]
+pub fn create_genesis_request(
+    address: [u8; 32],
+    genesis_validators: HashMap<contract_ffi::value::account::PublicKey, contract_ffi::value::U512>,
+) -> (GenesisRequest, HashMap<SystemContractType, WasmiBytes>) {
+    let genesis_account_addr = address.to_vec();
+    let mut contracts: HashMap<SystemContractType, WasmiBytes> = HashMap::new();
+
+    let initial_tokens = {
+        let mut ret = BigInt::new();
+        ret.set_bit_width(512);
+        ret.set_value("100000000000".to_string());
```

Comment:
I remember there was a small change in this line in #973 (this file probably was made before #973 was merged)

```suggestion
        ret.set_value(format!("{}", GENESIS_INITIAL_BALANCE));
```